functorch/functorch/csrc/BatchRulesLinearAlgebra.cpp

```diff
@@ -176,6 +182,181 @@ householder_product_batch_rule(const Tensor &input, c10::optional<int64_t> input
   return std::make_tuple(at::linalg_householder_product(input_, tau_), 0);
 }
 
+template <char const *op_name, typename A, A a, typename C>
+struct LinalgCheckMatrixUnaryRuleHelper;
+
+template <char const *op_name, typename F, F Func, typename A, typename... T>
+struct LinalgCheckMatrixUnaryRuleHelper<op_name, F, Func, typelist<A, T...>> {
+  static inline Tensor check_and_reshape_input(const Tensor& tensor, optional<int64_t> batch_dim) {
+    TORCH_CHECK(tensor.dim() > 2, op_name, ": The input tensor A must have at least 2 dimensions.");
+    return moveBatchDimToFront(tensor, batch_dim);
+  }
+
+  static oneOutput apply_one(
```

Comment:
(no action required) In c++17 we can create a single function that takes the number of returns as a template argument, using constexpr_if. e.g. https://stackoverflow.com/questions/59046161/how-to-change-the-return-type-depending-on-the-input-c . Unfortunately pytorch is still on c++14